src/map_objects.py

```diff
@@ -39,8 +43,16 @@ def __getitem__(self, gid):
         try:
             return self._objects[gid]
         except KeyError as e:
-            raise SyntaxError(
+            raise KeyError(
                 f"No object with gid {gid}.\nCheck if you "
                 f"defined a valid hitbox on the object you "
                 f"tried to access."
             ) from e
+
+    def get(self, gid):
+        try:
+            return self._objects[gid]
+        except KeyError:
+            return MapObjectType(
+                gid=gid, image=self._tilemap.get_tile_image_by_gid(gid)
+            )
```

Comment:
Could just use `self._objects.get` instead of a `try-except` block here.

```suggestion
    def get(self, gid):
        return self._objects.get(
            gid,
            MapObjectType(
                gid=gid,
                image=self._tilemap.get_tile_image_by_gid(gid)
            )
        )
```

Reply:
Oh, you're right! Don't know why I didn't think of that 😅